### DISCUSSION TITLE
Does Supabase realtime uses OT operations?

### DISCUSSION THREAD

**Discussion by mohammed-bahumaish** [supabase/supabase#14219] [Questions] [answered]
I have a Realtime application that i need to implement a realtime colaboration in.
which has a large json object that has to be synced between users in a room, OT operations looks good to sync the lagre Json object by sending only the changes between the users without sending the whole json object. 
so my question is how does superbase realtime handle that. does it send the whole json or broadcast only the changes?

**Comment by GaryAustin1** [comment]
If you are talking about realtime on the database (postgres_changes) it sends the entire row.

**Reply by mohammed-bahumaish** [reply]
so is it possible to send changes only with supabase realtime?
or i should implement that myself and send changes in channel?

**Reply by GaryAustin1** [reply, marked as answer]
https://supabase.com/docs/guides/realtime/extensions/presence
You might able to use the presence methods to keep things in sync on a object.  But I don't use that method, so can't say for sure based on what you describe you want to do.   You'll have to dig deeper into that, or see if someone familiar with it comes along.

**Reply by mohammed-bahumaish** [reply]
@GaryAustin1 Thanks.

`Presence utilizes an in-memory conflict-free replicated data type (CRDT) to track and synchronize shared state in an eventually consistent manner. It computes the difference between existing state and new state changes and sends the necessary updates to clients via Broadcast.`

i think that's what i'm looking for.
